medusa-config.js

```diff
@@ -1,133 +1,14 @@
-    break;
-  case "development":
-  default:
-    ENV_FILE_NAME = ".env";
-    break;
-}
+loadEnv(process.env.NODE_ENV, process.cwd())
 
-try {
-  dotenv.config({ path: process.cwd() + "/" + ENV_FILE_NAME });
-} catch (e) {}
-
-// CORS when consuming Medusa from admin
-// Medusa's docs are added for a better learning experience. Feel free to remove.
-const ADMIN_CORS = `${
-  process.env.ADMIN_CORS?.length
-    ? `${process.env.ADMIN_CORS},`
-    : "http://localhost:7000,http://localhost:7001,"
-}https://docs.medusajs.com,https://medusa-docs-v2-git-docs-v2-medusajs.vercel.app,https://medusa-resources-git-docs-v2-medusajs.vercel.app`;
-
-// CORS to avoid issues when consuming Medusa from a client
-// Medusa's docs are added for a better learning experience. Feel free to remove.
-const STORE_CORS = `${
-  process.env.STORE_CORS?.length
-    ? `${process.env.STORE_CORS},`
-    : "http://localhost:8000,"
-}https://docs.medusajs.com,https://medusa-docs-v2-git-docs-v2-medusajs.vercel.app,https://medusa-resources-git-docs-v2-medusajs.vercel.app`;
-
-const DATABASE_URL =
-  process.env.DATABASE_URL || "postgres://localhost/medusa-starter-default";
-
-const REDIS_URL = process.env.REDIS_URL || "redis://localhost:6379";
-
-const plugins = [];
-
-const modules = {
-  [Modules.CACHE]: true,
-  [Modules.EVENT_BUS]: true,
-  [Modules.AUTH]: true,
-  [Modules.USER]: {
-    resolve: "@medusajs/user",
-    options: {
-      jwt_secret: process.env.JWT_SECRET ?? "supersecret",
-    },
-  },
-  [Modules.FILE]: {
-    resolve: "@medusajs/file",
-    options: {
-      providers: [
-        {
-          resolve: "@medusajs/file-local-next",
-          options: {
-            config: {
-              local: {},
-            },
-          },
-        },
-      ],
-    },
-  },
-  [Modules.WORKFLOW_ENGINE]: true,
-  [Modules.STOCK_LOCATION]: true,
-  [Modules.INVENTORY]: true,
-  [Modules.PRODUCT]: true,
-  [Modules.PRICING]: true,
-  [Modules.PROMOTION]: true,
-  [Modules.CUSTOMER]: true,
-  [Modules.SALES_CHANNEL]: true,
-  [Modules.CART]: true,
-  [Modules.REGION]: true,
-  [Modules.API_KEY]: true,
-  [Modules.STORE]: true,
-  [Modules.TAX]: true,
-  [Modules.CURRENCY]: true,
-  [Modules.PAYMENT]: true,
-  [Modules.ORDER]: true,
-  [Modules.FULFILLMENT]: {
-    resolve: "@medusajs/fulfillment",
-    options: {
-      providers: [
-        {
-          resolve: "@medusajs/fulfillment-manual",
-          options: {
-            config: {
-              manual: {},
-            },
-          },
-        },
-      ],
-    },
-  },
-};
-
-/** @type {import('@medusajs/medusa').ConfigModule["projectConfig"]} */
-const projectConfig = {
-  databaseUrl: DATABASE_URL,
+module.exports = defineConfig({
+  databaseUrl: process.env.DATABASE_URL,
   http: {
-    storeCors: STORE_CORS,
-    adminCors: ADMIN_CORS,
-    authCors: process.env.AUTH_CORS || ADMIN_CORS,
+    storeCors: process.env.STORE_CORS,
+    adminCors: process.env.ADMIN_CORS,
+    authCors: process.env.AUTH_CORS,
```

Comment:
Should we keep default values for these? No strong opinions here just wanna hear some thoughts

Reply:
Not sure. Don't have any strong opinions either. If others think it should be defined here, I can update the code

Reply:
Do we have default values for these somewhere else, if not here? We will definitely need to have the CORS configured by default

Reply:
I think they are in the env file in theory no? at least they could be in the env template and env file

Reply:
<img width="773" alt="Screenshot 2024-05-30 at 09 48 58" src="https://github.com/medusajs/medusa-starter-default/assets/25098370/60d079ae-4821-4296-bbab-e9711d50dcb6">

Reply:
also, should we merge postgres_rul and database_url?

Reply:
Okay, let's leave it like this for now and see if someone gets confused. We can always add them back.